### DISCUSSION TITLE
Show more component/functionality

### DISCUSSION THREAD

**Discussion by elmaxe** [mantinedev/mantine#1047] [Feature requests]
### Usage example, including component/hook API

When a text is too long, a "Show more" button should appear. If you click it the whole content is shown.

![image](https://user-images.githubusercontent.com/33149910/155998683-cb2b1a39-2971-4945-9816-a22b50b7b76d.png)

Could be added to `Text`, or as a separate component.

```tsx
<ShowMore maxLength={120}>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi ipsum lectus, aliquam consequat urna faucibus, cursus vestibulum lacus. Pellentesque lobortis rhoncus egestas. Nulla ultricies turpis lacus, ultricies luctus augue bibendum mattis.
</ShowMore>
```

### Do you want to contribute this feature and create a pull request

Yes

**Comment by rtivital** [comment]
We already have a Spoiler component. If you decide to contribute it:
- Use dev branch as target
- Name component TextSpoiler
- Place component as subfolder of Spoiler
- Add demos to Spoiler component page
